### ISSUE TITLE
Ensure "good defaults" are consistent, e.g. in datatoken creation

### ISSUE THREAD

**Issue by trentmc** [oceanprotocol/ocean.py#1182]
### Background / Motivation

Q (Calina): related to defaults: there’s a big difference between our tests between Datatoken creation when creating assets and when “just” deploying Datatokens. Namely for the publish_market_order_fee_token (should it be ZERO_ADDRESS or OCEAN address?). same for publish_market_order_fee_address (should it be the publisher wallet address, or ZERO_ADDRESS?)

A (Trent): Great Q. It's useful to be consistent. Perhaps create an issue for it, and document the things that need consistency fixed. I can iterate with you on good defaults.

### TODOs

- [ ] Document which defaults need consistency fixed. E.g. as a GDoc or GSheet
- [ ] For each item, Trent & Calina agree on how to fix. Document in the GDoc / GSheet
- [ ]  Implement it

**Comment by calina-c** [comment]
@trentmc The only notable differences between what we employ in asset creation vs. single datatoken creation are the two I mentioned:

- publish_market_fee_token is OCEAN_address for asset creation and ZERO_ADDRESS for datatoken creation
- publish_market_order_fee_token is the publisher wallet address in asset creation and ZERO_ADDESS for datatoken creation

There is another difference in most tests, where we add the consumer_wallet as the fee manager, even if we use another publisher, but I think that is fine, since there's no way of knowing the consumer beforehand, so we should always pass it in as a parameter when needed (see implementation for more context).

Where it should be implemented:
- see if the fee_manager = consumer_wallet is really required in all the tests where it is used. My suspicion is it was just copied around and it's not necessary in ALL places where it is set. We should simplify where possible, while keeping it where needed.
- add the proper good defaults in arguments.py
- in tests, where we define the specific order fees and tokens which have become "good defaults", remove the explicit parameter setting
- iterate on datatoken creation in tests and turn it into a fixture. Possibilities: two types of DT in fixtures, or keep just one if we align the "good defaults".

**Comment by trentmc** [comment]
For defaults:
- publish_market_fee_token - I recommend ZERO_ADDRESS. Why: most use cases, especially when from ocean.py, and when people are getting going, aren't going to have a publish market fee. Secondly: I want to move away from OCEAN for fees and payment everywhere; it's lower friction for users to use DAI, USDC, WETH. (FYI having it for fees & payment doesn't help OCEAN token health. Ref Burniske MV=PQ.)
- publish_market_order_fee_token - ZERO_ADDRESS here too. Why: same as above

In tests, most of the time, fees should be zero, unless we're explicitly testing for fees. Then, we test well.

Re "consumer_wallet", one more note: labeling it "consumer_wallet" for a fee manager is confusing. Every time I see this, it annoys me. 
- To fix: have another account as pytest fixture called "fee_mgr_wallet", and use that instead of "consumer_wallet" 

> iterate on datatoken creation in tests and turn it into a fixture. Possibilities: two types of DT in fixtures, or keep just one if we align the "good defaults".

FYI, I re-introduced `DT()` as a fixture a couple days ago, along with `data_nft()` and `data_NFT_and_DT()`. Into conftest_ganache.py, [here](https://github.com/oceanprotocol/ocean.py/blob/4fed0f0fca78db6b7ef16eb3f44ab8b1331c527c/conftest_ganache.py#L156). You can leverage those. BTW if you want to rename `data_nft` to `data_NFT` go for it:)

**Comment by calina-c** [comment]
As #1220 has worked out, I realised our tests were using a lot of boilerplate/copy-paste even when not needed. I.e. even when not testing consumer/market balances, we were specifying addresses for fees and such. After #1220 is merged, I think this issue becomes moot, because:
- I removed any unnecessary setting of fees where we are specifically not testing fees, but other things
- I added defaults in FeeTokenArguments classes and DatatokenArguments, as needed
- The only places where defaults were unaligned were publish_market_order_fee_token and publish_market_order_fee_address, which are now inside FeeArguments
